pkg/scheduler/cache/cache.go

```diff
@@ -1247,10 +1261,27 @@ func (sc *SchedulerCache) GetMetricsData() {
 func (sc *SchedulerCache) setMetricsData(usageInfo map[string]*schedulingapi.NodeUsage) {
 	sc.Mutex.Lock()
 	defer sc.Mutex.Unlock()
+	beforeNodeinfo := make(map[string]string, len(sc.Nodes))
+	for nName, nInfo := range sc.Nodes {
```

Comment:
The pr is target to fix the panic issue. do not optimize code in this pr. 
Also please avoid doing the loop for non-important thing. It's might be a large cluster. And multiple go routines share the cache. There is a lock in this function. Make the code simple enough.

Reply:
i will keep it in mind, simplify the code for logging